ci/docker/runtime_functions.sh

```diff
@@ -399,6 +399,14 @@ unittest_ubuntu_gpu_R() {
     make rpkgtest R_LIBS=/tmp/r-site-library R_GPU_ENABLE=1
 }
 
+unittest_ubuntu_cpu_julia06() {
+    set -ex
+    julia -e 'versioninfo()'
+    julia -e 'Pkg.clone("julia", "MXNet")'
+    julia -e 'Pkg.build("MXNet")'
```

Comment:
I add a env var block for build in Jenkins config, 
```
withEnv(['MXNET_HOME=/workspace/ut-julia-cpu', 'JULIA_PKGDIR=/tmp/julia-pkg']) {
   ...
}
```
This build script will use the prebuilt binary if available.
See https://github.com/dmlc/MXNet.jl/blob/master/deps/build.jl#L11

Reply:
These scripts run within a docker environment. The mxnet root directory is available under /work/mxnet. In general, please don't specify env vars in the Jenkinsfile. Instead, use this shellscript (export...).

What exactly are the dependencies here? Is this something we could bake into the Dockerfile?

Could you elaborate "if available"? What happens, if it is not available?

Reply:
oh...what's the different between `/workspace/ut-julia-cpu` and `/work/mxnet`?

> What exactly are the dependencies here? Is this something we could bake into the Dockerfile?

All dependencies are Julia stuffs: https://github.com/dmlc/MXNet.jl/blob/master/REQUIRE.
Managing them via Julia's package manager is enough. There is nothing to do with Dockerfile, I think.

>Could you elaborate "if available"? What happens, if it is not available?

The build script will search the dir `MXNET_HOME/lib` first. If found a `libmxnet.so` under that dir, it will try to load it via `dlopen`. If `dlopen` failed, that file will be ignored.
In case of not available, it will build stuffs from source.

Reply:
changed in 1e0926e

Reply:
```/workspace/ut-julia-cpu``` is the path used on our Jenkins Slave host. ```/work/mxnet``` is the same directory, but mapped inside the Docker container.  

The question here is whether it makes sense to install these dependencies every single time during runtime. Instead, we could make use of the docker cache.

Reply:
Oh, Julia's deps follow the manner of rolling update. 
So, if the cache can be updated regularely (with function call `Pkg.update()`), I think its okay.
Which option do I need to enable for cache and cache update?

Reply:
There is no cache during runtime. Every container is starting from a clean state - the only actions that are being cached are the ones defined in the Dockerfiles. This means that every time you call ```Pkg.update()```, it will download and install the dependencies and throw everything away after the run has been finished. Any dependency management should happen in the Dockerfile (or the underlying install shellscripts).

Reply:
> Any dependency management should happen in the Dockerfile (or the underlying install shellscripts).

In case of Python, when will you update deps via `pip`?

Reply:
I'm afraid I don't understand your question. We specify the dependencies in our install scripts and then keep them consistent until we decide to change a version in the install script.

Reply:
I added an installation script for `julia` compiler (30aa3ca). (but not sure which version of Julia Ubuntu have.)

----

> This means that every time you call `Pkg.update()`, it will download and install the dependencies and throw everything away after the run has been finished.

The file `REQUIRE` is a Julia version of `requirements.txt`.
And all the Julia deps is listed in that file, end-user will use it / any PR can change it.
So, I want deps being installed during a normal build process.

Reply:
You can install these requirements as part of the docker build process. In order to make sure it is being re-run when the requirements.txt is changed, just add an ```ADD``` command to the Dockerfile before the script to install these dependencies is being called. This makes sure the underlying layers are automatically invalidated when the file is changed.

Reply:
Please check https://github.com/apache/incubator-mxnet/pull/10149/commits/701aca97df3414a0a3b98945c65529acc1d55c2a out.